src/core/search/lexer.lex

```diff
@@ -37,6 +38,7 @@ sq    \'
 esc_chars ['"\?\\abfnrtv]
 esc_seq \\{esc_chars}
 term_char [_]|\w
+term_char_ext {term_char}|\\[,.<>{}\[\]\\\"\':;!@#$%^&*()\-+=~\/ ]
```

Comment:
I know that this is what RedisSearch parser does but I think we should be more restrictive. The original request was for `-`, was not it? Is there a reason for a term to have `:;!@#$%^&*` ?

Reply:
in general, there are quoted strings that accept a wider range of characters, and unquoted strings that should be more limited.

Reply:
see here:
https://chatgpt.com/share/625da4c8-9991-4269-85c2-eec13dce1dae

Reply:
Other characters were asked in this comment https://github.com/dragonflydb/dragonfly/issues/3258#issuecomment-2292055368

Reply:
ok, and why do we need both `term_char_ext` and `term_char` ?

Reply:
fyi, our users are not responsible for product -we are. So not everything that is requested we do.  these decisions are critical because it's hard to revert them in the future.

Reply:
> term_char_ext is used only for tags, as I understand other parts of the expression can't use these escaping

I think we are aligned regarding the intent. Currently nothing limits using `term_char_ext` in other contexts. To fix this, you should disambiguate at grammar level  - to introduce TAGVAL terminal and use it when defining `tag_list_element`